### PR TITLE
Change Payflow 3DS params pass thru

### PR DESCRIPTION
**Pull request by filipebarcos**
We're changing this to align with https://github.com/activemerchant/active_merchant/wiki/Standardized-3DS-Fields

### CODE REVIEW

File: lib/active_merchant/billing/gateways/payflow.rb [hunk 1]
```diff
@@ -261,20 +261,7 @@ def add_credit_card(xml, credit_card, options = {})
           xml.tag! 'ExpDate', expdate(credit_card)
           xml.tag! 'NameOnCard', credit_card.first_name
           xml.tag! 'CVNum', credit_card.verification_value if credit_card.verification_value?
-
-          if options[:three_d_secure]
-            three_d_secure = options[:three_d_secure]
-            xml.tag! 'BuyerAuthResult' do
-              xml.tag! 'Status', three_d_secure[:status] unless three_d_secure[:status].blank?
```
[the remark this answers is not in the document]
Comment (bayprogrammer) [reply lib/active_merchant/billing/gateways/payflow.rb]: @filipebarcos @therufs So I was just looking into this a bit more, hoping against hope that maybe we could find a way to pass 3DS2 auth data, and I found this:

> ## How to send 3-D Secure authentication data
>
> After you've integrated with an MPI and can use the plug-in for cardholder
> authentication, you send the data you receive back from the MPI to Payflow
> during a sale or authorization request. The information you pass to Payflow
> [varies depending on whether the cardholder is enrolled in a 3-D Secure program or not](https://developer.paypal.com/docs/classic/payflow/3d-secure-mpi/#varying-fields).
> Use the developer documentation provided by your MPI to map the Payflow
> fields to the MPI-returned fields.
>
> # Payflow fields
>
> Field | Description | Data type/max length
> -- | -- | --
> `AUTHENTICATION_ID` | Unique identifier for this authentication validation. | alphanumeric, 64
> `AUTHENTICATION_STATUS` | Value returned by MPI indicating if authentication was successful, attempted or failed. | numeric, 1
> `CAVV` | Cardholder authentication verification value, also known as AAV. The value generated by the card-issuing bank proving the cardholder has been authenticated with a particular transaction. Returned if the AUTHENTICATION_STATUS is Successful or Attempted. | alphanumeric, 64
> `ECI` | E-Commerce Indicator. The ECI value indicates the level of security supported by the merchant when the cardholder provides payment card data for online purchase. | numeric, 1
> `XID` | 3-D Secure transaction ID. Returned if Successful or Attempted. Required. | alphanumeric, 64
> `THREEDSVERSION` | This field is for 3-D Secure 2.0. Contains the 3-D Secure version that was used to process the transaction. Possible values: 1.0.22.1.02.2.0Default = 1.0.2 | alphanumeric, 10
> `DSTRANSACTIONID` | This field is for 3-D Secure 2.0. Unique transaction identifier assigned by the Directory Server (DS) to identify a single transaction. Conditional. Note: Required for MasterCard Identity Check transaction in Authorization. | alphanumeric, 36
>
> **Note:** PayPal will implement 3-D Secure 2.0 in **September 2019**. While you can
> pass `THREEDVERSION` and `DSTRANSACTIONID` now, PayPal ignores these values until
> 3-D Secure 2.0 is implemented.

Source: https://developer.paypal.com/docs/classic/payflow/3d-secure-mpi/#how-to-send-3-d-secure-authentication-data

That actually sounds pretty promising! Am I reading things correctly there? Do you think that maps to the Payflow Pro API we currently support in ActiveMerchant?

[the remark this answers is not in the document]
Comment (filipebarcos) [reply lib/active_merchant/billing/gateways/payflow.rb]: I could check with PayPal,  but I believe that is for the "DoDirect" API, not for this one :(

[the remark this answers is not in the document]
Comment (filipebarcos) [reply lib/active_merchant/billing/gateways/payflow.rb]: I believe this is what we tried to do here and sorta halted https://github.com/activemerchant/active_merchant/pull/3285

[the remark this answers is not in the document]
Comment (bayprogrammer) [reply lib/active_merchant/billing/gateways/payflow.rb]: If you do have a contact at PayPal and don't mind double-checking, I'd sleep easier knowing for sure if they don't support it for this API.

Their 3DS docs I was looking at were kind of mixed in with the rest of the Payflow Pro docs I was looking at, but it wasn't super clear to me how it all fits together.

[the remark this answers is not in the document]
Comment (bayprogrammer) [reply lib/active_merchant/billing/gateways/payflow.rb]: Ah, sorry I didn't get your message about #3285 until after I wrote my last comment. Thanks for all the extra info!

**Pull request closed without merging**